Automattic/wp-calypso
# Limited GS on Personal: Load styles for treatment group (take 2)

Take 2 of https://github.com/Automattic/wp-calypso/pull/78589 which had to be reverted because of some problems in the Phabricator diff that was created to deploy the ETK changes. This PR includes these two additional fixes to avoid such problems:

- Prevent fatal by checking first if the site owner can be determined.
- Prevent uncached queries by caching the experiment assignment.

It also includes whether the site owner is assigned to the treatment group in the `global-styles/status` API endpoint so we can use that info in Calypso to properly develop the A/B experiences.

---

Fixes https://github.com/Automattic/dotcom-forge/issues/2781
Fixes https://github.com/Automattic/dotcom-forge/issues/2853

## Proposed Changes

Prevents the Global Styles from being blocked if the user is assigned to the treatment group of the `calypso_global_styles_personal` experiment.

## Testing Instructions

- Apply these changes to your sandbox.
- Sandbox the API.
- Create a new WP.com site and upgrade to the Personal plan.
- Sandbox the new site.
- Go to Appearance > Editor.
- Open the Styles section and select a custom style variation.
- Save the changes.
- Assign yourself to the control group of the experiment (see PCYsg-Fq7-p2#abacus-assignment-bookmarklet).
- Go to the [WP.com API console](https://developer.wordpress.com/docs/api/console/) and make a `GET` request to `wpcom/v2/sites/:site/global-styles/status`.
- Make sure that `globalStylesInPersonalPlan` is `false`.
- Visit the frontend of your site.
- Make sure you see a notice indicating that custom styles won't be public to visitors.
- Assign yourself to the treatment group of the experiment now.
- Remove the cached experiment assignment: ` wp_cache_delete( 'global-styles-on-personal-<SITE_ID>', 'a8c_experiments' );`
- Go to the [WP.com API console](https://developer.wordpress.com/docs/api/console/) and make a `GET` request to `wpcom/v2/sites/:site/global-styles/status`.
- Make sure that `globalStylesInPersonalPlan` is `true` now.
- Visit the frontend of your site.
- Make sure you no longer see the Limited Global Styles notice.
